metriq_gym/cli.py

```diff
@@ -12,58 +12,71 @@ def parse_arguments() -> argparse.Namespace:
         Parsed arguments as an argparse.Namespace object.
     """
     parser = argparse.ArgumentParser(description="Metriq-Gym benchmarking CLI")
-    parser.add_argument(
+    subparsers = parser.add_subparsers(dest="action", required=True, help="Action to perform")
+
+    # Subparser for dispatch.
+    dispatch_parser = subparsers.add_parser("dispatch", help="Dispatch jobs")
+    dispatch_parser.add_argument(
         "input_file",
         type=str,
         help="Path to the file containing the benchmark parameters",
     )
-    parser.add_argument(
+    dispatch_parser.add_argument(
         "-n", "--num_qubits", type=int, default=8, help="Number of qubits (default is 8)"
     )
-    parser.add_argument(
+    dispatch_parser.add_argument(
         "-s", "--shots", type=int, default=8, help="Number of shots per trial (default is 8)"
     )
-    parser.add_argument(
+    dispatch_parser.add_argument(
         "-t", "--trials", type=int, default=8, help="Number of trials (default is 8)"
     )
-    parser.add_argument(
+    dispatch_parser.add_argument(
         "-p",
         "--provider",
         type=str,
         choices=["ibmq", "quantinuum", "ionq"],
         default="ibmq",
         help="String identifier for backend provider service",
     )
-    parser.add_argument(
+    dispatch_parser.add_argument(
         "-b",
         "--backend",
         type=str,
         default="qasm_simulator",
         help='Backend to use (default is "qasm_simulator")',
     )
-    parser.add_argument(
+    dispatch_parser.add_argument(
         "-c",
         "--confidence_level",
         type=float,
         default=0.025,
         help="p-value confidence level to use (default is 0.025)",
     )
-    parser.add_argument(
+    dispatch_parser.add_argument(
         "-j",
         "--jobs_file",
         type=str,
         default=DEFAULT_JOBS_FILE,
         help="File in local directory where async jobs are recorded",
     )
-    parser.add_argument(
+    dispatch_parser.add_argument(
         "--token", type=str, help="IBM Quantum API token (must be supplied to run on real hardware)"
     )
-    parser.add_argument(
+    dispatch_parser.add_argument(
         "--instance",
         type=str,
         help="Name of the IBM Quantum plan instance (e.g. 'ibm-q/open/main')",
     )
 
-    args = parser.parse_args()
+    # Subparser for poll.
+    poll_parser = subparsers.add_parser("poll", help="Poll jobs")
+    poll_parser.add_argument(
```

Comment:
nit: `jobs_file` parser is already in dispatch parser; I wonder if there is a way to specify it only once

Reply:
That's true, we could just specify this in the global parser and not twice in the subsequent dispatch and poll parsers. Done in 63b8ed3.